### DISCUSSION TITLE
QML/UI framework reusability

### DISCUSSION THREAD

**Discussion by DragoonX6** [musescore/MuseScore#16459] [Q&A]
According to [Tantacrul's video about the making of version 4](https://youtu.be/Qct6LKbneKQ?t=581) the component library is open source you encourage other applications to use it (previous link is timestamped to that quote). I'm interested in using it, but looking through it quickly it doesn't seem like the UI libraries are fully decoupled from the rest of the framework.

To what extent is the UI framework coupled to the rest of the framework? If I want to separate the UI parts for use in my completely unrelated application (different field entirely), how hard is that? Is there interest in splitting this off as an official musescore organization library?

Full disclosure: I'm not yet sure if the UI library is right for me, and if I want to use one of the built-in QML material design themes or not.

**Comment by ottergauze** [comment]
I apologise for replying to this after a few years but this is something I'd really like to see get developed more, including a potential for documentation on how to properly utilise it. I think Musescore's new UI and the design work behind it is great (and is the only cross-platform GUI app that's not web-based that doesn't make me retch when I look at it) and I think it'd be a disservice for work like that on an open source project to not make it easy for others to pick up and use it.

**Comment by anabelle2001** [comment]
same here!

if it's possible to release the UI component items under a more permissive licence (MIT / LGPL?), I'd love to see if I can use them at work. No worries if they have to stay GPL.
